### PR TITLE
Front-End of Editors, /doc/. Minor changes to Model

### PR DESCRIPTION
**Pull request by arunabh98**
I have made the front end of editors of both Markdown and WSYIWYG. Apart from this I have modified the front end of /doc/. Made some changes to Model which makes it now possible to upload documents and even view them after clicking on them.

### CODE REVIEW

**Comment by sb0210** [comment]
Arunabh, nice work  :).
Before merging,  remove all the virtual environment files and the binaries. If you have used any python library, put it into requirements/common.txt. Also if you are using any css/js files directly from the web, download them locally and then use it. 
Look at the comments by landscape (for code quality). And also check if the build is working on travis.

File: docupload/templates/docupload/markdown.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{% extends "docupload/base.html" %}
```
Comment (udiboy1209) [inline docupload/templates/docupload/markdown.html:1]: This file has a lot of mixed indentation, stick to tab=4 spaces. There might be very easy ways to fix this whole files indentation based on your editor

File: docupload/templates/docupload/index.html [hunk 2]
```diff
@@ -1,15 +1,57 @@
...
-    {% endfor %}
-    </ul>
-{% else %}
-    <p>No docs are available.</p>
-{% endif %}
+{% extends "docupload/base.html" %}
+
+{% block style %}
+	body {
+		background: black;
```
Comment (udiboy1209) [inline docupload/templates/docupload/index.html]: This file should also not use tabs, but 4 spaces instead

File: docupload/templates/docupload/editor_choice.html [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{% extends "docupload/base.html" %}
```
Comment (udiboy1209) [inline docupload/templates/docupload/editor_choice.html]: Same here, convert to tabs to 4 spaces

File: docupload/migrations/0002_auto_20160505_0834.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
```
Comment (udiboy1209) [inline docupload/migrations/0002_auto_20160505_0834.py:1]: shouldn't this file be gitignored @siddharthbulia @DheerendraRathor ?

File: docupload/htmlify.py [hunk 5]
```diff
@@ -3,6 +3,7 @@
 '''
 
 import os
+
```
Comment (udiboy1209) [inline docupload/htmlify.py:6]: random change :stuck_out_tongue: ?

**Comment by udiboy1209** [comment]
Nice work @Arunabh98 ! You've used tabs almost everywhere instead of 4 spaces. Please convert them.  I'll test this soon enough once you have it ready.

**Comment by udiboy1209** [comment]
Great work @Arunabh98 :tada: :smile:

**Comment by arunabh98** [comment]
Thanks man!! Tune bahut madad ki meri
Thanks everybody
My first accepted PR :)

**Pull request merged**